packages/keystone/fetch.js

```diff
@@ -145,36 +144,39 @@ const fetchWithRetriesAndLogger = async (url, options = {}) => {
     let lastResponse
     // At least one request on maxRetries = 0
     do {
+        let timeout
         try {
             const controller = new AbortController()
             const signal = controller.signal
-            const response = await Promise.race([
-                fetchWithLogger(url, { ... fetchOptions, signal }, { skipTracingHeaders, skipXTargetHeader }),
-                new Promise((_, reject) =>
-                    setTimeout(() => {
-                        controller.abort()
-                        reject(new Error('Abort request by timeout'))
-                    }, abortRequestTimeout)
-                ),
-            ])
+
+            timeout = setTimeout(() => {
```

Comment:
I guess you don't need timeout anymore. AbortController should do its job

Reply:
Refactored